cmsutils.py

```diff
@@ -70,6 +70,35 @@ def _memorySizeGB():
     return count
 
 
+def _gpuList(gpu_type):
+    """
+    Returns the number of supported GPUs for the given type ('cuda' or 'rocm').
+
+    :param gpu_type: Type of GPU to check ('cuda' or 'rocm')
+    :return: Number of supported GPUs
+    """
+    if gpu_type == "cuda":
+        cmd = "cudaComputeCapabilities"
+    elif gpu_type == "rocm":
+        cmd = "rocmComputeCapabilities"
+    else:
+        raise ValueError("Unsupported GPU type: {0}".format(gpu_type))
+
+    e, o = run_cmd(cmd)
+
+    if e != 0:
+        return []
+
+    supported_lines = [
+        "{0}:{1}".format(gpu_type, i)
+        for i, line in enumerate(o.strip().splitlines())
+        if "unsupported" not in line
+    ]
+    return supported_lines
+
+
+CUDAGPUList = _gpuList("cuda")
+ROCMGPUList = _gpuList("rocm")
```

Comment:
@iarspider , do not create these global variables ( this means every time this file is imported then we run `xxxComputeCapabilities`). Instead change the name of `_gpuList()` and directly call it from jobscheduler.py

Reply:
Ok